crates/service-client/src/lib.rs

```diff
@@ -34,38 +35,49 @@ mod options;
 mod proxy;
 
 #[derive(Debug, Clone)]
-pub struct ServiceClient {
+pub struct ServiceClient<B> {
     // TODO a single client uses the pooling provided by hyper, but this is not enough.
     //  See https://github.com/restatedev/restate/issues/76 for more background on the topic.
-    http: HttpClient,
+    http: HttpClient<B>,
     lambda: LambdaClient,
 }
 
-impl ServiceClient {
-    pub(crate) fn new(http: HttpClient, lambda: LambdaClient) -> Self {
+impl<B> ServiceClient<B> {
+    pub(crate) fn new(http: HttpClient<B>, lambda: LambdaClient) -> Self {
         Self { http, lambda }
     }
 }
 
-impl ServiceClient {
+impl<B> ServiceClient<B>
```

Comment:
Why do we need ServiceClient to be generic over body?

Reply:
Because the request body is generic. In discovery it's going to be `Full` probably while in the invoker is going to use `BodyStream<ReceiverStream<>>` (and probably in future something custom).

But anyway, this PR is far from being ready, so i'm not sure this is right anyway :smile: